public_html/wp-content/plugins/wordcamp-coming-soon-page/classes/wordcamp-coming-soon-page.php

```diff
@@ -92,6 +92,7 @@ public function render_dynamic_styles() {
 			return;
 		}
 
+		// phpcs:ignore WordPress.PHP.DontExtract.extract_extract -- Not sure whats the alternative to this could be.
```

Comment:
🤔 

That's a good question.

I'd be tempted to remove the use of `extract`, since it's a violation of the guidelines. https://core.trac.wordpress.org/ticket/22400 has more discussion and a link to an article.

We don't have to do that now, though. For now I'd just remove the `phpcs:ignore`, and let the warning remain so we can fix it in the future.

On the other hand, this does kind of feel like an appropriate usage of `extract()`, since it keeps the logic DRY (it's used by multiple templates). One way to keep it and avoid the problems with it would be to add phpdoc to declare the variables at the point where they're `extract()`ed, but then those docs wouldn't be DRY.

🤔 

I don't have a good answer. 🤷‍♂️ 

Maybe it's fine to leave the `phpcs:ignore`? Whatever you think is best works for me.

Reply:
I am going to keep this for now. But lets re visit when we have some time. Adding a todo here.

Reply:
👍